src/lib/work_selector/offset.ml

```diff
@@ -0,0 +1,35 @@
+open Core_kernel
+
+module Make
+    (Inputs : Intf.Inputs_intf)
+    (Lib : Intf.Lib_intf with module Inputs := Inputs) =
+struct
+  (* Offset will be generated randomly at each execution, and will stay
+     constant for the duration of the execution *)
+  let offset =
+    (* Self init makes Random.int produce a new number at each execution *)
+    Random.self_init () ;
```

Comment:
we already call this function once at the program entrypoint, so we shouldn't need to do it again. 
[see](https://github.com/MinaProtocol/mina/blob/develop/src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml#L1991)

Reply:
For this to work, `offset` probably needs to become `lazy`. Otherwise, this will be called before we hit the `Random.self_init` call in the entrypoint.

Reply:
I didn’t manage to get fresh offset at each execution with `lazy` (calling `Lazy.force inside the `work` function)

Reply:
Outdated, the randomness does not need to be fresh for the unit tests.

Reply:
During testing i.e running a snark worker I saw fresh offsets at each run.